camel/configs.py

```diff
@@ -11,11 +11,14 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 # =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+from __future__ import annotations
+
 from abc import ABC
 from dataclasses import asdict, dataclass, field
-from typing import Any, Dict, List, Optional, Sequence, Union
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Sequence, Union
```

Comment:
@ocss884 Thanks. Could you explain why is only `OpenAIFunction` included in "TYPE_CHECKING"? Since other types are either not used during runtime.

Reply:
Cuz it is the culprit who triggers a circular import if we're going to remove all the imports from the top level \__init\__. Not all imports for type hints purpose would cause such a problem.

To be short:
`camel.configs` imports `OpenAIFunction` which would load `camel.functions` module. In the `functions.search_functions` file it imports `camel.agents` and in `camel.agents.chat_agents` it has a dependency from `camel.configs`. Then there is a circular import. (to be more specific another reason is all imports are from .. import)

The reason we saw no bug before is that the top level \__init\__ have a `import camel.agents` first. But since we'd like to improve the import time for the CLI module, we have to clean these useless top level imports otherwise as I described in the issue all commands would suffer a significant lag around 0.45s due to the import even without executing anything specific. The problem described above is a typical circular import case caused by type hint (here it is `OpenAIFunction`) and one of the standard ways to solve it is to use `TYPE_CHECKING` to remove these imports from runtime

Reply:
Thank you for your explanation and I get your point. But it seems in `camel/agents/chat_agent.py`, line `from typing import Any, Callable, Dict, List, Optional, Tuple` is included in the `if TYPE_CHECKING:`. Would it be better if we unify the behavior to put all the basic types inside or outside `TYPE_CHECKING`?

Reply:
> Thank you for your explanation and I get your point. But it seems in `camel/agents/chat_agent.py`, line `from typing import Any, Callable, Dict, List, Optional, Tuple` is included in the `if TYPE_CHECKING:`. Would it be better if we unify the behavior to put all the basic types inside or outside `TYPE_CHECKING`?

Surely it is better. My fault I forgot to move it outside.